File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/RailBot.java
package Auxiliary_Modules;

import LivinGrimoire.Kokoro;

public class RailBot {
    private final EventChatV2 ec;
    private String context = "stand by";
    private ElizaDBWrapper elizaWrapper = null; // Starts null (no DB)

    public RailBot(int limit) {
        ec = new EventChatV2(limit);
    }

    public RailBot() {
        this(5);
    }
    /**
     * Enables database features. Must be called before any save/load operations.
     * If never called, RailBot works in memory-only mode.
     */
    public void enableDBWrapper() {
        if (elizaWrapper == null) {
            elizaWrapper = new ElizaDBWrapper();
        }
    }
    public void disableDBWrapper() {
        elizaWrapper = null;
    }

    public void setContext(String context) {
        if (context.isEmpty()) {
            return;
        }
        this.context = context;
    }

    private String respondMonolog(String ear) {
        if (ear.isEmpty()) {
            return "";
        }
        String temp = ec.response(ear);
        if (!temp.isEmpty()) {
            this.context = temp;
        }
        return temp;
    }

    public void learn(String ear) {
        if (ear.isEmpty() || ear.equals(this.context)) {
            return;
        }
        ec.addKeyValue(this.context, ear);
        this.context = ear;
    }

    public String monolog() {
        return respondMonolog(this.context);
    }

    public String respondDialog(String ear) {
        return ec.response(ear);
    }
    public String respondLatest(String ear) {
        return ec.responseLatest(ear);
    }

    public void learnKeyValue(String context, String reply) {
        ec.addKeyValue(context,reply);
    }

    public void feedKeyValuePairs(java.util.List<AXKeyValuePair> kvList) {
        if (kvList.isEmpty()) {
            return;
        }
        for (AXKeyValuePair kv : kvList) {
            learnKeyValue(kv.getKey(), kv.getValue());
        }
    }
    // save/load functionalities
    public void saveLearnedData(Kokoro kokoro){
        if (elizaWrapper == null) {return;}
        elizaWrapper.sleepNSave(ec, kokoro);
    }
    private String loadableMonologMechanics(String ear, Kokoro kokoro) {
        // loads data if available
        if (ear.isEmpty()) {
            return "";
        }
        String temp = elizaWrapper.respond(ear,ec,kokoro);
        if (!temp.isEmpty()) {
            this.context = temp;
        }
        return temp;
    }
    public String loadableMonolog(Kokoro kokoro) {
        if(elizaWrapper == null){return monolog();}
        return loadableMonologMechanics(this.context,kokoro);
    }
    public String loadableDialog(String ear, Kokoro kokoro) {
        if(elizaWrapper == null){return respondDialog(ear);}
        return elizaWrapper.respond(ear,ec,kokoro);
    }

}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/RailBot.kt
package auxiliary_modules

import livinGrimoire.Kokoro

class RailBot @JvmOverloads constructor(limit: Int = 5) {
    private val ec: EventChatV2
    private var context: String = "stand by"
    private var elizaWrapper: ElizaDBWrapper? = null // Starts null (no DB)

    init {
        ec = EventChatV2(limit)
    }

    /**
     * Enables database features. Must be called before any save/load operations.
     * If never called, RailBot works in memory-only mode.
     */
    fun enableDBWrapper() {
        if (elizaWrapper == null) {
            elizaWrapper = ElizaDBWrapper()
        }
    }

    fun disableDBWrapper() {
        elizaWrapper = null
    }

    fun setContext(context: String) {
        if (context.isEmpty()) {
            return
        }
        this.context = context
    }

    private fun respondMonolog(ear: String): String {
        if (ear.isEmpty()) {
            return ""
        }
        val temp = ec.response(ear)
        if (!temp.isEmpty()) {
            context = temp
        }
        return temp
    }

    fun learn(ear: String) {
        if (ear.isEmpty() || ear == context) {
            return
        }
        ec.addKeyValue(context, ear)
        context = ear
    }

    fun monolog(): String {
        return respondMonolog(context)
    }

    fun respondDialog(ear: String): String {
        return ec.response(ear)
    }

    fun respondLatest(ear: String): String {
        return ec.responseLatest(ear)
    }

    fun learnKeyValue(context: String, reply: String) {
        ec.addKeyValue(context, reply)
    }

    fun feedKeyValuePairs(kvList: List<AXKeyValuePair>) {
        if (kvList.isEmpty()) {
            return
        }
        for (kv in kvList) {
            learnKeyValue(kv.key, kv.value)
        }
    }

    // save/load functionalities
    fun saveLearnedData(kokoro: Kokoro) {
        if (elizaWrapper == null) {
            return
        }
        elizaWrapper!!.sleepNSave(ec, kokoro)
    }

    private fun loadableMonologMechanics(ear: String?, kokoro: Kokoro): String {
        // loads data if available
        if (ear!!.isEmpty()) {
            return ""
        }
        val temp = elizaWrapper!!.respond(ear, ec, kokoro)
        if (!temp.isEmpty()) {
            context = temp
        }
        return temp
    }

    fun loadableMonolog(kokoro: Kokoro): String {
        return if (elizaWrapper == null) {
            monolog()
        } else loadableMonologMechanics(context, kokoro)
    }

    fun loadableDialog(ear: String, kokoro: Kokoro): String {
        return if (elizaWrapper == null) {
            respondDialog(ear)
        } else elizaWrapper!!.respond(ear, ec, kokoro)
    }
}